app/models/discourse_single_sign_on.rb

```diff
@@ -203,10 +221,36 @@ def change_external_attributes_and_override(sso_record, user)
       end
     end
 
+    profile_background_missing = !user.user_profile.profile_background.present?
```

Comment:
No need for this variable. What you are looking for is `user.user_profile.profile_background.blank?`